framework/src/utils/MooseAppCoordTransform.C

```diff
@@ -528,6 +528,10 @@ MultiAppCoordTransform::setDestinationCoordTransform(
 {
   _destination_app_transform = &destination_app_transform;
 
+  // Assume user knows what they are doing if they have multiple coordinate systems
```

Comment:
```suggestion
  // Don't error check mismatching coordinate system types if we've been asked to skip coordinate collapsing since in that case the mismatch doesn't matter
```